gemfirexd/tools/src/test/java/com/pivotal/gemfirexd/jdbc/transactions/snapshot/SnapshotTransactionTest.java

```diff
@@ -1011,6 +1119,416 @@ public void run() {
+    int numRow = 0;
+    while (rs.next()) {
+      numRow++;
+      assertEquals("Primary Key coloumns should be 10 ", 10, rs.getInt(1));
+      assertEquals("Second columns should be 10 , ", 10, rs.getInt(2));
+    }
+    assertEquals("ResultSet should have two rows ", 1, numRow);
+    txMgrImpl.commit();
+
+    rs.close();
+    st2.close();
+    conn.commit();
+    conn.close();
+  }
+
+  public void testConflictWrite() throws Exception {
+    Connection conn = getConnection();
+    Statement st = conn.createStatement();
+    ResultSet rs;
+    int numRows = 0;
+    st.execute("Create table t1 (c1 int not null , c2 int not null, primary key(c1)) replicate persistent enable concurrency checks " + getSuffix());
+    conn.commit();
+    GemFireCacheImpl.getInstance().getCacheTransactionManager().begin(IsolationLevel.SNAPSHOT, null);
+    for (int i = 0; i < 10; i++) {
+      st.execute("insert into t1 values (" + i + ", " + i + ")");
+    }
+
+    Cache cache = Misc.getGemFireCache();
+    final TXManagerImpl txMgrImpl = (TXManagerImpl)cache
+        .getCacheTransactionManager();
+    this.gotConflict = false;
+
+    Thread thread = new Thread(new Runnable() {
+      @Override
+      public void run() {
+        try {
+          Connection newConn = TestUtil.getConnection();
+          Statement st2 = newConn.createStatement();
+          txMgrImpl.begin(IsolationLevel.SNAPSHOT, null);
+          try {
+            for (int i = 8; i < 10; i++) {
+              st2.execute("insert into t1 values (" + i + ", " + i + ")");
+            }
+            fail("expected a conflict here");
+          } catch (ConflictException ce) {
+            gotConflict = true;
+          } catch (SQLException se) {
+            if (!(se.getCause() instanceof ConflictException)) {
+              threadEx = se;
+              se.printStackTrace();
+            } else {
+              txMgrImpl.rollback();
+              newConn.rollback();
+            }
+          }
+        } catch (Throwable t) {
+          threadEx = t;
+          fail("unexpected exception", t);
+        }
+      }
+    });
+    thread.start();
+    thread.join();
+    GemFireCacheImpl.getInstance().getCacheTransactionManager().commit();
+    conn.commit();
+
+    GemFireCacheImpl.getInstance().getCacheTransactionManager().begin(IsolationLevel.SNAPSHOT, null);
+    rs = st.executeQuery("Select * from t1 where t1.c2 >= 0");
+    numRows = 0;
+    while (rs.next()) {
+      numRows++;
+    }
+    assertEquals("ResultSet should contain 10 rows ", 10, numRows);
+    Misc.getGemFireCache().getCacheTransactionManager().commit();
+    conn.commit();
+
+  }
+
+  public void testSnapshotGet() throws Exception {
+    Connection conn = getConnection();
+    Statement st = conn.createStatement();
+    ResultSet rs;
+    int numRows = 0;
+    st.execute("Create table t1 (c1 int not null , c2 int not null, primary key(c1)) replicate persistent enable concurrency checks " + getSuffix());
```

Comment:
All your test is using replicate persistent. Partiton persistent should also be added. Isn't it?